### ISSUE TITLE
Jump to code does not work within a code block

### ISSUE THREAD

**Issue by ghost** [Enter-tainer/typst-preview#182]
**Describe the bug**
When clicking in the preview window we cannot jump to the code when the cursor is within a code block.

    #lorem(50)
    
    ```rust
    we cannot jump to the source code in a code block...
    1
    2
    3
    4
    5
    ```
    
    #lorem(50)

**Expected behavior**
The jumping behaviour should work like in normal text.

**Package/Software version:**

VSCode version(Help -> About):
```plain
Version: 1.84.2 (user setup)
Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
Date: 2023-11-09T10:51:52.184Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045
```

**Comment by Myriad-Dreamin** [comment]
The span information is provided by typst compiler, and it doesn't provide a such information for code blocks.

We may convert it to a upstream issue when we double confirmed that.

**Comment by Enter-tainer** [comment]
Currently jump to source and vice versa heavily relies on upstream and we can only do it in best effort. And for typst, there is not a good way to universally determine the mapping between source code and rendering result.

**Comment by gfpj** [comment]
I understand, thanks for the answers.
Maybe it is possible to jump to the end of the last known position (the lorem call before). That would not be a perfect solution but definetly good enough.

**Comment by Myriad-Dreamin** [comment]
@gfpj fixed by https://github.com/typst/typst/pull/2674

**Comment by Dherse** [comment]
Indeed, the best we can do is provide the span of the code block itself but we can't provide information about where *inside* of the block you are, this is due to the following:

- Code blocks can be created from a regular string in which case we don't have its span, perhaps in the future we might have the ability to have a `Spanned` string that is produced both in code and when using the `read` function, that does sound to me like a good solution
- We don't keep Span information for raw block syntax since it gets turned into a regular string (something that could also be solved by my previous statement)

**Comment by Enter-tainer** [comment]
This is fixed in 0.11.0
